lib/src/native/realm_core.dart

```diff
@@ -187,6 +197,9 @@ class _RealmCore {
             nullptr,
           );
         }
+        if (config.migrationCallback != null) {
+          _realmLib.realm_config_set_migration_function(configHandle._pointer, Pointer.fromFunction(migration_callback, false), config.toWeakHandle(), nullptr);
```

Comment:
Isn't it better to use the approach with passing scheduler, similar to errorHandlerUserdata below in the same method?

Reply:
We know that the migration function is going to be executed on the caller thread, so we don't need to wrap the Core callback in our own that dispatches to the isolate thread.